rdflib/term.py

```diff
@@ -1641,16 +1641,17 @@ def _parseXML(xmlstring: str) -> xml.dom.minidom.Document:  # noqa: N802
 def _parseHTML(htmltext: str) -> xml.dom.minidom.DocumentFragment:  # noqa: N802
     try:
         import html5lib
-
-        parser = html5lib.HTMLParser(tree=html5lib.treebuilders.getTreeBuilder("dom"))
-        retval = parser.parseFragment(htmltext)
-        retval.normalize()
-        return retval
     except ImportError:
         raise ImportError(
             "HTML5 parser not available. Try installing"
             + " html5lib <http://code.google.com/p/html5lib>"
         )
+    parser = html5lib.HTMLParser(tree=html5lib.treebuilders.getTreeBuilder("dom"))
+    retval = parser.parseFragment(htmltext)
+    if parser.errors:
+        raise ValueError("html5lib failed to parse.", htmltext)
+    retval.normalize()
+    return retval
```

Comment:
I looked at the html5lib code and I think it may be slightly better to run the parser in strict mode, in which case it will throw by itself.

```suggestion
    parser = html5lib.HTMLParser(tree=html5lib.treebuilders.getTreeBuilder("dom"), strict=True)
    retval = parser.parseFragment(htmltext)
    retval.normalize()
    return retval
```

For more info see [[ref](https://html5lib.readthedocs.io/en/latest/html5lib.html#html5lib.html5parser.HTMLParser)]